src/client/entity-editor/series-section/actions.ts

```diff
@@ -81,6 +86,52 @@ export function addSeriesItem(data: Relationship): Action {
 	};
 }
 
+
+/**
+ * This action creator first sorts the series items object and then pass the
+ * sorted object in the payload while dispatching the action.
+ *
+ * @param {number} oldIndex - Old Position of the series item.
+ * @param {number} newIndex - New Position of the series item.
+ * @returns {void}
+ */
+export function sortSeriesItems(oldIndex, newIndex):any {
+	return (dispatch, getState) => {
+		const state = getState();
+		const seriesItems = state.get('seriesSection').get('seriesItems');
+		const orderTypeValue = state.get('seriesSection').get('orderType');
+		const seriesItemsObject = seriesItems.toJS();
+		const seriesItemsArray = Object.entries(seriesItemsObject);
+		const automaticSort = [];
+		let automaticSortedArr: [string, Relationship][]; // stores the sorted array of series items(sorting performed on number)
+
+		if (orderTypeValue === 1) { // OrderType 1 for Automatic Ordering
+			seriesItemsArray.forEach((seriesItem:[string, Relationship]) => {
+				seriesItem[1].attributes.forEach((attribute: Attribute) => {
+					if (attribute.attributeType === 2) { // Attribute Type 2 for number
+						automaticSort.push({number: attribute.value.textValue, seriesItem});
+					}
+				});
+			});
+			automaticSort.sort(sortRelationshipOrdinal('number')); // sorts the array of series items on number attribute
```

Comment:
We'll have to have another look at this sorting function to implement a natural sort order; currently it will do a language sort, so it will sort for example 1 > 10 > 100 > 2

![image](https://user-images.githubusercontent.com/6179856/129772959-7d15f998-9883-41eb-a81d-6d40b373c148.png)

This can be done in another PR

Reply:
See PR #691